### PR TITLE
[XFileShare] Add NxLoad

### PR DESCRIPTION
**Pull request by kr4ssi**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x ] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [ x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x ] Improvement
- [x ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

### CODE REVIEW

File: youtube_dl/extractor/xfileshare.py [hunk 1]
```diff
@@ -58,10 +58,12 @@ class XFileShareIE(InfoExtractor):
         (r'vidshare\.tv', 'VidShare'),
         (r'vup\.to', 'VUp'),
         (r'xvideosharing\.com', 'XVideoSharing'),
+        (r'nxload\.com', 'NxLoad'),
+
     )
 
     IE_DESC = 'XFileShare based sites: %s' % ', '.join(list(zip(*_SITES))[1])
-    _VALID_URL = (r'https?://(?:www\.)?(?P<host>%s)/(?:embed-)?(?P<id>[0-9a-zA-Z]+)'
+    _VALID_URL = (r'https?://(?:www\.)?(?P<host>%s)/(?:embed-|embed\/)?(?P<id>[0-9a-zA-Z]+)'
```
Comment (dstftw) [inline youtube_dl/extractor/xfileshare.py]: 1. DRY.
2. No need to escape slash.

File: youtube_dl/extractor/xfileshare.py [hunk 2]
```diff
@@ -58,10 +58,12 @@ class XFileShareIE(InfoExtractor):
         (r'vidshare\.tv', 'VidShare'),
         (r'vup\.to', 'VUp'),
         (r'xvideosharing\.com', 'XVideoSharing'),
+        (r'nxload\.com', 'NxLoad'),
+
```
Comment (dstftw) [inline youtube_dl/extractor/xfileshare.py]: Remove empty line.

File: youtube_dl/extractor/xfileshare.py [hunk 3]
```diff
@@ -151,7 +163,7 @@ def _real_extract(self, url):
         if not formats:
             urls = []
             for regex in (
-                    r'(?:file|src)\s*:\s*(["\'])(?P<url>http(?:(?!\1).)+\.(?:m3u8|mp4|flv)(?:(?!\1).)*)\1',
+                    r'(?:file|src)\s*:\s*(["\\\']{1,2})(?P<url>http(?:(?!\1).)+\.(?:m3u8|mp4|flv)(?:(?!\1).)*)\1',
```
Comment (dstftw) [inline youtube_dl/extractor/xfileshare.py]: Incorrect. Should only match `'` or `\'` as quote.

**Comment by 3dik** [comment]
Today, I released an own, independent nxload extractor. Your code seems to be far superior. However, there might still be a tiny chance that you find some useful hints in my code. So [here it is](https://git.sr.ht/~edik/nxloaddecrypter). ;)

**Pull request closed without merging**